tests/test_agent_manager.py

```diff
@@ -384,6 +456,18 @@ async def dummy_status():
     }
     assert_equal_ish(shouldbe, all_agents, sortby=["environment", "name"])
 
+    start = "agent2"
+    code, all_agents = await am.list_agents(env=None, start=start)
+    assert code == 200
+    for a in all_agents["agents"]:
+        assert a["name"] > start, f"List of agent should not contain a name (={a['name']}) before or equal to start (={start})"
+
+    end = "agent2"
+    code, all_agents = await am.list_agents(env=None, end=end)
+    assert code == 200
+    for a in all_agents["agents"]:
+        assert a["name"] < end, f"List of agent should not contain a name (={a['name']}) after or equal to end (={end})"
+
```

Comment:
Also assert the number of records here. If the result is an empty list, this test will always succeed.

Reply:
Did it in https://github.com/inmanta/inmanta/pull/2521/commits/3ef9c64b173a26c0fa2feebd57a595f17006ed2b